src/langauge-servers/javaUtils.ts

```diff
@@ -0,0 +1,187 @@
+            process.env.ProgramW6432 && path.join(process.env.ProgramW6432, "Java"), // Oracle JDK per machine
+            process.env.ProgramFiles && path.join(process.env.ProgramFiles, "AdoptOpenJDK"), // AdoptOpenJDK per machine
+            process.env.ProgramW6432 && path.join(process.env.ProgramW6432, "AdoptOpenJDK"), // AdoptOpenJDK per machine
+            path.join(localAppDataFolder, "Programs", "AdoptOpenJDK"), // AdoptOpenJDK per user
+        ].filter(Boolean) as string[];
+        const jvmStores = _.uniq(possibleLocations);
+        jvmStores.forEach(jvmStore => {
+            const jvms = fs.readdirSync(jvmStore);
+            jvms.forEach(jvm => {
+                const javaLoc = path.join(jvmStore, jvm);
+                if (fs.existsSync(javaLoc)) {
+                    javaLocations.push(javaLoc)
+                }
+            })
+        })
+    }
+
+    // common place for Linux
+    if (isLinux) {
+        const jvmStore = "/usr/lib/jvm";
+        const jvms = fs.readdirSync(jvmStore);
+
+        jvms.forEach(jvm => {
+            const javaLoc = path.join(jvmStore, jvm);
+            if (fs.existsSync(javaLoc)) {
+                javaLocations.push(javaLoc)
+            }
+        })
+
+    }
+
+    return javaLocations;
+}
+
+
+export const getJavaVersion = (javaPath: string) => {
+    let javaVersion = checkVersionInReleaseFile(javaPath);
+    if (!javaVersion) {
+        javaVersion = checkVersionByCLI(javaPath);
+    }
+    return javaVersion;
+}
+
+const checkVersionInReleaseFile = (javaPath: string): number => {
+    if (!javaPath) {
+        return 0;
+    }
+    const releaseFile = path.join(javaPath, "release");
+    if (!fs.existsSync(releaseFile)) {
+        return 0;
+    }
+
+    const content = fs.readFileSync(releaseFile, { encoding: 'utf-8' });
+    const regexp = /^JAVA_VERSION="(.*)"/gm;
+    const match = regexp.exec(content.toString());
+    if (!match) {
+        return 0;
+    }
+    const majorVersion = parseMajorVersion(match[1]);
+    return majorVersion;
+}
+
+const parseMajorVersion = (version: string): number => {
+    if (!version) {
+        return 0;
+    }
+    // Ignore '1.' prefix for legacy Java versions
+    if (version.startsWith("1.")) {
+        version = version.substring(2);
+    }
+    // look into the interesting bits now
+    const regexp = /\d+/g;
+    const match = regexp.exec(version);
+    let javaVersion = 0;
+    if (match) {
+        javaVersion = parseInt(match[0]);
+    }
+    return javaVersion;
+}
+
+const checkVersionByCLI = (javaHome: string): number => {
+    if (!javaHome) {
+        return 0;
+    }
+
+    const javaBin = path.join(javaHome, "bin", JAVA_FILENAME);
+    let stdout = ""
+    try{
+        const stdout = cp.execFileSync(javaBin, ["-version"])
+    } catch (e) {
+        throw new Error("Java home location is invalid");
+    }
+
+    const regexp = /version "(.*)"/g;
+    const match = regexp.exec(stdout);
+    if (!match) {
```

Comment:
should probably add a log here? otherwise if this won't work in the wild we wouldn't be able to resolve